### DISCUSSION TITLE
Question forum

### DISCUSSION THREAD

**Discussion by ashleychontos** [ashleychontos/pySYD#37] [Q&A]
Do you have a general support question or are you experiencing a problem not directly associated with software malfunction? (i.e. need help asteroseismology-ing?) This is the place for you!

**Comment by bmorris3** [comment]
Hi @ashleychontos, @danxhuber suggested I give pysyd a spin for some non-Kepler light curves I'm working on. Is there a way to supply pysyd with custom, non-Kepler, offline light curves? Thanks!

**Reply by ashleychontos** [reply]
Sounds like a great idea @bmorris3! 

Could you first clarify what you mean by 'custom' and 'offline' light curves? In theory, any light curve should work assuming that the units are correct, of course. It might not optimized for say, irregularly sampled data with large gaps but again, it should still work and these are actually great things to test, characterize and quantify (if possible).

**Reply by bmorris3** [reply]
In my case, I've got simulated light curves with oscillations and granulation. I have time and flux arrays that I'd like to hand off to pysyd.

**Reply by ashleychontos** [reply]
This actually sounds like a great test for `pysyd`, as we have not yet extensively tested the "light curve only" input mode. 

I think we have a couple options:
- If you send me a single light curve, I am happy to closely monitor the software myself to 1) test if it works in this mode (as I said, this is very new) and 2) see how it affects the overall outputs (primarily the computation time). 
- You're also free to have a go at it yourself - as long as the time array is days and flux array is normalized flux. The function that computes the power spectrum currently assumes these units and normalizes the power spectrum accordingly. Of course the end goal is to interface with `astropy.units` so we don't have to assume any units.

We could also do both, I'm just curious to see how it does! BTW are we talking of order like tens, thousands or millions of simulated light curves? I'm trying to determine the best mode for you to run these, whether it be on a semi-individual basis or in a batch mode.

**Reply by bmorris3** [reply]
I'd love to have a go myself, but I haven't seen an example in the docs that illustrates how to do this. Might there be an example that I missed?

**Reply by ashleychontos** [reply]
The documentation could definitely benefit from more examples! (and I appreciate the feedback so I'll be sure to add something about this)

It's essentially the same as if you had both a _LC.txt and _PS.txt file. The software will determine what inputs you have and processes it as far/best as it can go i.e. if it only finds a light curve, it will automatically calculate a power spectrum and subsequently saves it for later iterations in the same directory as the light curve (as opposed to calculating a new one every time). It should then run through the global analysis, as usual.

**Reply by ashleychontos** [reply]
and of course, please let me know if you run into any issues!

**Reply by bmorris3** [reply]
Ok so to confirm – I can place a two-column `_LC.txt` file in `pysyd/data/`, containing time (days) and flux (normalized to unity)?

**Reply by ashleychontos** [reply]
yep, that sounds right to me!

**Reply by bmorris3** [reply]
Hi @ashleychontos, 

I've given this a shot by creating a `4243803_LC.txt` file and then running pysyd via CLI as suggested by the docs. I'm hitting an error that looks like this: 
```python
$ pysyd run --star 4243803 -dv --ux 5000 --mc 200
Traceback (most recent call last):
  File "/Users/bmmorris/miniconda3/envs/research/bin/pysyd", line 8, in <module>
    sys.exit(main())
  File "/Users/bmmorris/miniconda3/envs/research/lib/python3.9/site-packages/pysyd/cli.py", line 663, in main
    args.func(args)
  File "/Users/bmmorris/miniconda3/envs/research/lib/python3.9/site-packages/pysyd/pipeline.py", line 198, in run
    params = utils.Parameters(args=args)
  File "/Users/bmmorris/miniconda3/envs/research/lib/python3.9/site-packages/pysyd/utils.py", line 116, in __init__
    self.add_cli(args)
  File "/Users/bmmorris/miniconda3/envs/research/lib/python3.9/site-packages/pysyd/utils.py", line 367, in add_cli
    self.check_cli(args)
  File "/Users/bmmorris/miniconda3/envs/research/lib/python3.9/site-packages/pysyd/utils.py", line 406, in check_cli
    raise InputError("\nWhen running multiple stars via command line, the number \n of values provided for %s MUST equal the number of stars\n" % each)
pysyd.utils.InputError: <InputError>
```
Any ideas?

**Reply by ashleychontos** [reply]
Hi @bmorris3 many thanks for this. I think I have an idea what's happening, since I had to recently change how some of the variables are passed with the required JOSS testing. 

Would you mind dropping that light curve in the thread so I can better debug from my end? That would be really helpful (or alternatively, you can email me the LC if you prefer). Thanks!

**Reply by danxhuber** [reply]
@ashleychontos the crash also happens with the default examples, so seems unrelated to the input file format. I'll submit an issue since seems to be a bug that affects software functionality. Note this only seems to happen with -- keywords only, i.e. 
pysyd run --star 1435467 -dv 
works but 
pysyd run --star 1435467 -dv --ux 5000
crashes.

**Reply by ashleychontos** [reply]
This further supports the hypothesis about the way the parameters are passed but fortunately, should be an easy fix. I will work on this this week and let you know when it's been fixed. Thanks for bringing this up @bmorris3 @danxhuber
